lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -461,6 +470,20 @@ def check_watched_item(self, job_state):
             # job is no longer viable - remove from watched jobs
             return None
 
+    def _handle_unschedulable_job(self, job, job_state):
```

Comment:
Can you get in the habit of putting these things into pykube_util. I know you're not using the Pulsar Kubernetes runner yet, but I think long term it is will be a valuable tool and we should work to keep feature parity - or at least making it easy to have feature parity. I'm going to have to refactor this out of here for reuse in that code and there is not test coverage here so I risk breaking things - if it were developed to be used that way from the start it would be less likely to be broken when I refactor it.

Reply:
Done in https://github.com/galaxyproject/galaxy/pull/10496/commits/dc7d3fdf4226b44b8b929e419a2aa88de5ce25f7#diff-83b685d37c2473c87101dc9de4da0464b79bb2042539132e5c41ee7261e93b98R71